File: src/jvmMain/kotlin/domain/NormalSudokuGame.java
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

class NormalSudokuGame extends AbstractSudokuGame{
	public static final GameFactory.GameMode GAMEMODE = GameFactory.GameMode.NORMAL;
	
	public NormalSudokuGame(GameFactory.Size size, GameFactory.Difficulty difficulty) {
		super(size, difficulty);
	}

	@Override
	protected AbstractSudokuSolver initiateSolver(SudokuField solutionField) {
		return new SudokuSolver(solutionField);
	}

	@Override
	protected AbstractSudokuGameFieldGenerator initiateGenerator(SudokuField gameField) {
		return new SudokuGameFieldGenerator(gameField);
	}
}

File: src/jvmMain/kotlin/domain/NormalSudokuGame.kt
import GameFactory.Difficulty

/**
 * the classical Version of a sudoku game extending the abstract SudokuGame
 */
internal class NormalSudokuGame(size: GameFactory.Size, difficulty: Difficulty) : AbstractSudokuGame(size, difficulty) {
    /**
     * instanciates the classical SudokuSolver
     * please find more information in the abstract sudoku game
     */
    override fun initiateSolver(solutionField: SudokuField): AbstractSudokuSolver {
        return SudokuSolver(solutionField)
    }

    /**
     * instanciates the classical SudokuGenerator
     * please find more information in the abstract sudoku game
     */
    override fun initiateGenerator(gameField: SudokuField): AbstractSudokuGameFieldGenerator {
        return SudokuGameFieldGenerator(gameField)
    }

    companion object {
        val GAMEMODE = GameFactory.GameMode.NORMAL
    }
}
